contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,166 @@
+<frontmatter>
+  title: Introduction to Angular
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Angular
+
+**Author: [Ronak Lakhotia](https://github.com/RonakLakhotia)**                                                              
+Reviewers: [Tan Heng Yeow](https://github.com/tanhengyeow), [Jacob Li PengCheng](https://github.com/jacoblipech)
+
+## What is Angular?
+
+Angular, according to the [official website](https://angular.io/docs), is described as follows:
+
+>Angular is a platform that makes it easy to build applications with the web. Angular combines declarative templates, dependency injection, end to end tooling, and integrated best practices to solve development challenges. Angular empowers developers to build applications that live on the web, mobile, or the desktop.
+
+Let us understand some of the key terms used in the above description with the help of a Hello World example:
```

Comment:
The flow is patchy. Consider something like this (outline only):

Let's look into what _declarative template_ means.
* Let's use an example of creating a trivial web app that behaves as follows {give screenshots, describe behavior}
* The _view_ part (i.e., the part users see) is defined as a file (Angular calls these _templates_) using HTML and additional HTML-like tags added by Angular. Here is the template for the example {give code here}
* The template is _declarative_ because it _describes_ the elements in the view rather than give instructions on how to construct the view. The code that has the behavior instructions is given outside the template, as a component (?) written in TypeScript language, which will look as follows. {give code here}
* [good to have, but can be omitted] In contrast, the code for the same using a competing technology jQuery(?) (which is not declarative) will look like the following. {give code here}

Now let's look at what _dependency injection_ means. { as this is an advanced technique, you can try to explain the objective of the technique rather than the mechanics }
* It is an advanced technique that allows ... so that ... can be done more easily.
* For example, it can be used to {give example from HelloWorld}